packages/core-utils/src/external/ethers/network.ts

```diff
@@ -0,0 +1,6 @@
+import { Provider } from '@ethersproject/abstract-provider'
+
+export const getChainId = async (provider: Provider): Promise<number> => {
+  const network = await provider.getNetwork()
+  return network.chainId
```

Comment:
You could consider returning an entire object.  What I recently added was a common interface to use for everything.

interface Chain {
  id: number
  chainName: string
  isTestChain: boolean
  isLocalChain: boolean
  multicallAddress?: string
  getExplorerAddressLink: (string) => string
  getExplorerTransactionLink: (string) => string
  isL1: boolean
  isL2: boolean
  get l1() { // returns a Chain object of the corresponding L1 in the L1 -L2 pairing }
  get l2() { // analagous to get l1() but I imagine in future it returns an array of multiple l2s}
}

Reply:
This is a cool idea, will do in a separate PR

Reply:
Yea the get l1() and get l2() on the frontend specifically really simplifies our code a ton where we get rid of branching like

const l2Balance = getBalance({chain: chainId === 1 ? 10 : chainId === kovan ? kovanOpt : chainId === 10 ? 10 : ...})

const l2Balance = getBalance({chain: chain.l2.id})